cpp/src/arrow/compute/kernels/scalar_if_else_benchmark.cc

```diff
@@ -386,8 +441,13 @@ BENCHMARK(CaseWhenBenchString)->Args({kFewItems, 99});
 BENCHMARK(CaseWhenBenchStringContiguous)->Args({kFewItems, 0});
 BENCHMARK(CaseWhenBenchStringContiguous)->Args({kFewItems, 99});
 
-BENCHMARK(CoalesceBench64)->Args({kNumItems, 0});
-BENCHMARK(CoalesceBench64)->Args({kNumItems, 99});
+BENCHMARK(CoalesceBench64)->Args({kNumItems, 0, 2});
+BENCHMARK(CoalesceBench64)->Args({kNumItems, 0, 4});
+BENCHMARK(CoalesceBench64)->Args({kNumItems, 99, 2});
+BENCHMARK(CoalesceBench64)->Args({kNumItems, 99, 4});
+
+BENCHMARK(CoalesceScalarBench64)->Args({kNumItems, 0});
+BENCHMARK(CoalesceScalarStringBench)->Args({kNumItems, 0});
 
 BENCHMARK(CoalesceNonNullBench64)->Args({kNumItems, 0});
 BENCHMARK(CoalesceNonNullBench64)->Args({kNumItems, 99});
```

Comment:
It is possible to also parametrize the null_probability used in the coalesce benchmarks? I think we have some logic to do this (see e.g. RegressionArgs and the Take benchmarks).